grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -0,0 +1,1539 @@
+    matchingEntitiesOnly?: boolean
+}
+
+interface EntityColorData {
+    color: Color
+    colorDomainValue: string
+}
+// Points used on the X axis
+interface SimplePoint {
+    value: number
+    entity: string
+    time: number
+}
+
+export interface SimpleChartSeries {
+    seriesName: string
+    points: SimplePoint[]
+}
+
+enum BarShape {
+    Bar,
+    BarPlaceholder,
+}
+
+interface Bar {
+    kind: BarShape.Bar
+    color: Color // color from the variable
+    seriesName: string
+    yPoint: StackedPoint<EntityName>
+}
+
+interface BarPlaceholder {
+    kind: BarShape.BarPlaceholder
+    seriesName: string
+}
+
+type BarOrPlaceholder = Bar | BarPlaceholder
+
+interface Item {
+    entityId: string
+    entityColor: EntityColorData | undefined
+    bars: Bar[] // contains the y values for every y variable
+    xPoint: SimplePoint // contains the single x value
+}
+
+interface PlacedItem extends Item {
+    pixelSpaceXOffset: number // x value (in pixel space) when placed in final sorted order and including shifts due to one pixel entity minimum
+}
+
+interface TooltipProps {
+    item: Item
+    highlightedSeriesName?: string
+    targetTime?: Time
+    timeColumn: CoreColumn
+    formatColumn: CoreColumn
+    xAxisColumn: CoreColumn
+}
+
+interface EntityWithSize {
+    entityId: string
+    xValue: number
+}
+interface LabelCandidate {
+    item: EntityWithSize
+    bounds: Bounds
+    isPicked: boolean
+    isSelected: boolean
+}
+
+interface LabelWithPlacement {
+    label: JSX.Element
+    preferredPlacement: number
+    correctedPlacement: number
+    labelKey: string
+}
+
+interface LabelCandidateWithElement {
+    candidate: LabelCandidate
+    labelElement: JSX.Element
+}
+
+const MARKER_MARGIN: number = 4
+const MARKER_AREA_HEIGHT: number = 25
+
+@observer
+export class MarimekkoChart
+    extends React.Component<{
+        bounds?: Bounds
+        manager: MarimekkoChartManager
+    }>
+    implements ChartInterface, HorizontalColorLegendManager, ColorScaleManager {
+    base: React.RefObject<SVGGElement> = React.createRef()
+
+    defaultBaseColorScheme = ColorSchemeName.continents
+    defaultNoDataColor = "#959595"
+    labelAngleInDegrees = -45 // 0 is horizontal, -90 is vertical from bottom to top, ...
+
+    transformTable(table: OwidTable): OwidTable {
+        if (!this.yColumnSlugs.length) return table
+        if (!this.xColumnSlug) return table
+        const { excludedEntities } = this.manager
+        const { yColumnSlugs, manager, colorColumnSlug, xColumnSlug } = this
+
+        if (excludedEntities) {
+            const excludedEntityIdsSet = new Set(excludedEntities)
+            table = table.columnFilter(
+                OwidTableSlugs.entityId,
+                (entityId) => !excludedEntityIdsSet.has(entityId as number),
+                `Excluded entity ids specified by author: ${excludedEntities.join(
+                    ", "
+                )}`
+            )
+        }
```

Comment:
Given that this now runs beyond ScatterPlot, it might make sense to move it to the Grapher, both to deduplicate implementation, and to very slightly improve performance.

I checked the database and no other chart except ScatterPlot has `excludedEntities` defined.

```sql
SELECT DISTINCT config->"$.type" 
FROM charts
WHERE config->"$.excludedEntities" IS NOT NULL
```